aaelf64-morello/aaelf64-morello.rst

```diff
@@ -0,0 +1,831 @@
+  has a different value.)
+
+- ``GOT`` is the address of the Global Offset Table, the table of code and data
+  addresses to be resolved at dynamic link time. The ``GOT`` and each entry in it
+  must be aligned to the pointer-size.
+
+- ``GDAT(S+A)`` represents a pointer_sized entry in the ``GOT`` for address
+  ``S+A``. The entry will be relocated at run time with relocation
+  ``R_MORELLO_GLOB_DAT(S+A)``.
+
+- ``G(expr)`` is the address of the GOT entry for the expression expr.
+
+- ``GTLSDESC(S+A)`` represents a consecutive pair of pointer-sized entries in
+  the GOT which contain a ``tlsdesc`` structure describing the thread-local
+  variable located at offset ``A`` from thread-local symbol ``S``. The first
+  entry holds a pointer to the variable's TLS descriptor resolver function and
+  the second entry holds a platform-specific offset or pointer. The pair of
+  pointer-sized entries will be relocated with ``R_MORELLO_TLSDESC(S+A)``.
+
+- ``TLSDESC(S+A)`` resolves to a contiguous pair of pointer-sized values, as
+  created by GTLSDESC(S+A).
+
+- ``CAP_INIT`` Generate a capability with all required information. When used on
+  its own represents the operations needs to be done for handling ``R_MORELLO_CAPINIT``.
+
+- ``CAP_SIZE`` The size of the underlying memory region that the capability can
+  reference. This may not directly map to the symbol size.
+
+- ``CAP_PERM`` The permission of the capability. This may not directly map to
+  the type of the symbol.
+
+- ``[msb:lsb]`` is a bit-mask operation representing the selection of bits in a
+  value. The bits selected range from ``lsb`` up to ``msb`` inclusive. For
+  example, ‘bits [3:0]’ represents the bits under the mask 0x0000000F. When
+  range checking is applied to a value, it is applied before the masking
+  operation is performed.
+
+.. topic:: pointer-size
+
+  The pointer-size is 64 bits for the A64 ABI and 128 bits for the pure capability (C64)
+  ABI.
+
+.. raw:: pdf
+
+   PageBreak
+
+
+Static Morello relocations
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. warning:: The ELF64 Code of the relocations are subject to change.
+
+.. class:: aaelf64-morello-static-relocations
+
+.. table:: Relocations to generate 19, 21, and 33 bit PC-relative addresses
+
+    +-------+-----------------------------------+-----------------+-------------------------------------------------------------------------+
+    | ELF64 | Name                              | Operation       | Comment                                                                 |
+    | Code  |                                   |                 |                                                                         |
+    +=======+===================================+=================+=========================================================================+
+    | 57348 | ``R_MORELLO_LD_PREL_LO17``        | ``S+A -``       | Set a load-literal immediate value to bits [20:4] of X;                 |
```

Comment:
Please end sentences starting with a Capital with a period`.` This also makes two sentences in a row hard to read like what happens in the rendering of the last row. I'm not sure if you meant to have two senteces separated by a return in the table follow eachother directly without a return, but that's how rst interprets it.